python/cudf/cudf/_lib/strings/json.pyx

```diff
@@ -1,18 +1,19 @@
-# Copyright (c) 2021, NVIDIA CORPORATION.
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.
 
 from libcpp.memory cimport unique_ptr
 from libcpp.utility cimport move
+from libcpp cimport bool
 
 from cudf._lib.column cimport Column
 from cudf._lib.cpp.column.column cimport column
 from cudf._lib.cpp.column.column_view cimport column_view
 from cudf._lib.cpp.scalar.scalar cimport string_scalar
-from cudf._lib.cpp.strings.json cimport get_json_object as cpp_get_json_object
+from cudf._lib.cpp.strings.json cimport (get_json_object as cpp_get_json_object, get_json_object_options)
 from cudf._lib.cpp.types cimport size_type
 from cudf._lib.scalar cimport DeviceScalar
 
 
-def get_json_object(Column col, object py_json_path):
+def get_json_object(Column col, object py_json_path, bool allow_single_quotes=False, bool strip_quotes_from_single_strings=True, bool missing_fields_as_nulls=False):
```

Comment:
This Cython API should mirror the C++ API, with this signature:
```suggestion
def get_json_object(Column col, object py_json_path, get_json_object_options options):
```

The construction of the `get_json_object_options` should be handled in a separate function.

Reply:
How can a Python consumer pass a C++ object?

Reply:
Sorry, I wasn't being clear about what I was thinking. This is what I had in mind: an API that acts as a pure wrapper around libcudf, and a separate Cython function to accept and parse the bools. I'm totally flexible on naming (`cpp_get_json_object` may not be a good name).

```python
def cpp_get_json_object(Column col, object py_json_path, get_json_object_options options):
    cdef unique_ptr[column] c_result

    cdef column_view col_view = col.view()
    cdef DeviceScalar json_path = py_json_path.device_value

    cdef const string_scalar* scalar_json_path = <const string_scalar*>(
        json_path.get_raw_ptr()
    )

    with nogil:
        c_result = move(cpp_get_json_object(
            col_view,
            scalar_json_path[0],
            options,
        ))

    return Column.from_unique_ptr(move(c_result))

def get_json_object(Column col, object py_json_path, bool allow_single_quotes=False, bool strip_quotes_from_single_strings=True, bool missing_fields_as_nulls=False):
    """
    Apply a JSONPath string to all rows in an input column
    of json strings.
    """
    cdef get_json_object_options options
    options.set_allow_single_quotes(allow_single_quotes)
    options.set_strip_quotes_from_single_strings(strip_quotes_from_single_strings)
    options.set_missing_fields_as_nulls(missing_fields_as_nulls)

    return cpp_get_json_object(col, py_json_path, options)
```

Reply:
Thanks @shwina! I agree your suggestion is clearer with where the line is drawn. And yes, this is motivated by similar considerations that are driving the proposal for a "pylibcudf" layer. The part I want to separate the most is the Python/Cython-level handling of `bool` options from the C++ API which passes a `get_json_object_options` to libcudf.